static/js/list_rendering.js

```diff
@@ -0,0 +1,104 @@
+var list_render = (function () {
+    var DEFAULTS = {
+        LOAD_COUNT: 20,
+        instances: {},
+    };
+
+    // @params
+    // container: jQuery object to append to.
+    // template_name: Handlebars template name.
+    // list: The list of items to progressively append.
+    // opts: An object of random preferences.
+    var func = function ($container, list, opts) {
+        var meta = {
+            offset: 0,
+            listRenders: {},
+        };
+
+        if (!opts) {
+            return;
+        }
+
+        var $nearestScrollingContainer = $container;
+        while ($nearestScrollingContainer.length) {
+            if ($nearestScrollingContainer.is("body, html")) {
+                blueslip.warn("Please wrap progressive scrolling lists in an element with 'max-height' attribute. Error found in:\n" + util.preview_node($container));
+                break;
+            }
+
+            if ($nearestScrollingContainer.css("max-height") !== "none") {
+                break;
+            }
+
+            $nearestScrollingContainer = $nearestScrollingContainer.parent();
+        }
+
+        var prototype = {
```

Comment:
I named it as prototype since it will work like one from an outside perspective, but no. It could however be made into one -- which may be a good use case as this will be called many times perhaps and prototypes consume lots less memory (as only one copy exists in memory).